pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -105,22 +106,49 @@ func ConvertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) (model
 		Hosts: []string{"*"},
 	})
 
-	virtualService := &networking.VirtualService{
-		Hosts:    []string{"*"},
-		Gateways: []string{model.IstioIngressGatewayName},
+	gatewayConfig := model.Config{
+		ConfigMeta: model.ConfigMeta{
+			Type:      model.Gateway.Type,
+			Group:     model.Gateway.Group,
+			Version:   model.Gateway.Version,
+			Name:      ingress.Name + "-" + model.IstioIngressGatewayName,
+			Namespace: model.IstioIngressNamespace,
+			Domain:    domainSuffix,
+		},
+		Spec: gateway,
```

Comment:
And when you add the gateway manually to helm, this entire function can go away.